File: plsql-squid/src/test/java/br/com/felipezorzo/sonar/plsql/api/LoopStatementTest.java
package br.com.felipezorzo.sonar.plsql.api;

import static org.sonar.sslr.tests.Assertions.assertThat;

import org.junit.Before;
import org.junit.Test;

public class LoopStatementTest extends RuleTest {

    @Before
    public void init() {
        setRootRule(PlSqlGrammar.LOOP_STATEMENT);
    }

    @Test
    public void matchesSimpleLoop() {
        assertThat(p).matches(""
                + "loop "
                + "null; "
                + "end loop;");
    }
    
    @Test
    public void matchesNestedLoop() {
        assertThat(p).matches(""
                + "loop "
                + "loop "
                + "null; "
                + "end loop; "
                + "end loop;");
    }

}

File: zpa-core/src/test/kotlin/org/sonar/plugins/plsqlopen/api/statements/LoopStatementTest.kt
/**
 * Z PL/SQL Analyzer
 * Copyright (C) 2015-2025 Felipe Zorzo
 * mailto:felipe AT felipezorzo DOT com DOT br
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
package org.sonar.plugins.plsqlopen.api.statements

import com.felipebz.flr.tests.Assertions.assertThat
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.sonar.plugins.plsqlopen.api.PlSqlGrammar
import org.sonar.plugins.plsqlopen.api.RuleTest

class LoopStatementTest : RuleTest() {

    @BeforeEach
    fun init() {
        setRootRule(PlSqlGrammar.LOOP_STATEMENT)
    }

    @Test
    fun matchesSimpleLoop() {
        assertThat(p).matches(""
                + "loop "
                + "null; "
                + "end loop;")
    }

    @Test
    fun matchesNestedLoop() {
        assertThat(p).matches(""
                + "loop "
                + "loop "
                + "null; "
                + "end loop; "
                + "end loop;")
    }

    @Test
    fun matchesLabeledLoop() {
        assertThat(p).matches(""
                + "<<foo>> loop "
                + "null; "
                + "end loop foo;")
    }

}
